src/domain/auth/auth.repository.interface.ts

```diff
@@ -18,6 +18,16 @@ export interface IAuthRepository {
   verifyToken(accessToken: string): AuthPayload;
 
   decodeToken(accessToken: string): AuthPayloadWithClaims<AuthPayload>;
+
+  isChain(args: {
+    chainId: string;
+    authPayload: AuthPayload | undefined;
+  }): boolean;
+
+  isSigner(args: {
+    signerAddress: `0x${string}`;
+    authPayload: AuthPayload | undefined;
+  }): boolean;
```

Comment:
Reviewing it once more and comparing it to the functionality provided by `IAuthRepository`, maybe this makes sense to be part of a class which implements `AuthPayload` wdyt? As in, we are directly comparing external data _against_ the state of `authPayload`.

```typescript
export class AuthPayload implements z.infer<typeof AuthPayloadSchema> {
  readonly chain_id: string;
  readonly signer_address: `0x${string}`;

  constructor(chain_id: string, signer_address: `0x${string}`) {
    this.chain_id = chain_id;
    this.signer_address = signer_address;
  }

  isForChain(chainId: string): boolean {
    return this.chain_id === chainId;
  }

  isForSigner(signerAddress: `0x${string}`): boolean {
    return this.signer_address.toLowerCase() === signerAddress.toLowerCase();
  }
}
```

Reply:
Great suggestion. I made the relevant adjustments in f3a0bee5201c5af0b21818233a35984c753fea62. Lmk what you think.